### PR TITLE
parser - Handle project name and version variables

### PR DESCRIPTION
**Pull request by josepht**
Replace $SNAPCRAFT_PROJECT_VERSION and $SNAPCRAFT_PROJECT_NAME
with their respective values from the origin's snapcraft.yaml
in remote parts definitions.

LP:#1619418

### CODE REVIEW

File: snapcraft/internal/parser.py [hunk 1]
```diff
@@ -134,7 +134,24 @@ def _update_source(part, origin):
     return part
 
 
-def _process_entry_parts(entry_parts, parts, origin, maintainer, description):
+def _replace_variables(item, name, version):
```
Comment (sergiusens) [inline snapcraft/internal/parser.py]: this looks a lot like `project_loader`'s `replace_attr`. Maybe generalize it a bit to define a list of replacement tuples such that you can pass something like this

```python
[
    ('$SNAPCRAFT_PROJECT_NAME', name),
    ('$SNAPCRAFT_PROJECT_VERSION', version),
]
```

and then in the function for the `str` instances:
```python
for replacement in replacements:
    attr.replace(replacement[0], replacement[1])
```

Comment (josepht) [inline snapcraft/internal/parser.py]: Done!

File: snapcraft/internal/parser.py [hunk 2]
```diff
@@ -134,7 +134,23 @@ def _update_source(part, origin):
     return part
 
 
-def _process_entry_parts(entry_parts, parts, origin, maintainer, description):
+def _replace_variables(item, replacements):
```
Comment (sergiusens) [inline snapcraft/internal/parser.py]: Hmm, there might have been a miscomm somewhere :)

The general idea was to remove the `_` from https://github.com/snapcore/snapcraft/blob/master/snapcraft/internal/project_loader.py#L201 and adapt that ;-)

**Comment by sergiusens** [comment]
retest this please

File: snapcraft/internal/parser.py (hunk 2 above)
Comment (josepht) [inline snapcraft/internal/parser.py]: Updated.

**Pull request merged**